spec/system/candidate_interface/automatically_carry_over_unsubmitted_application_to_new_cycle_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'rails_helper'
+
+RSpec.feature 'Automatically carry over unsubmitted applications' do
+  include CandidateHelper
+
+  around do |example|
+    Timecop.freeze(Date.new(2020, 8, 1)) do
+      example.run
+    end
+  end
+
+  scenario 'Carry over application and remove all application choices' do
+    given_i_am_signed_in_as_a_candidate
+    when_i_have_an_unsubmitted_application
+    and_the_recruitment_cycle_ends
+    and_the_unsubmitted_application_carry_over_worker_runs
+
+    when_i_sign_in_again
+    and_i_visit_the_application_dashboard
+    then_i_see_a_copy_of_my_application
+
+    when_i_view_referees
+    then_i_can_see_the_referees_i_previously_added
+
+    when_i_view_courses
+    then_i_can_see_that_i_need_to_select_courses
+
+    and_i_select_a_course
+    and_i_complete_the_section
+    and_i_submit_my_application
+    and_my_application_is_awaiting_references
+  end
+
+  def given_i_am_signed_in_as_a_candidate
+    @candidate = create(:candidate)
+    login_as(@candidate)
+  end
+
+  def when_i_have_an_unsubmitted_application
+    @application_form = create(
+      :completed_application_form,
+      submitted_at: nil,
+      candidate: @candidate,
+      with_gces: true,
+      safeguarding_issues_status: :no_safeguarding_issues_to_declare,
+    )
+    create(
+      :application_choice,
+      status: :unsubmitted,
+      application_form: @application_form,
+    )
+    @unrequested_references = create_list(:reference, 2, feedback_status: :not_requested_yet, application_form: @application_form)
+  end
+
+  def and_the_recruitment_cycle_ends
+    Timecop.safe_mode = false
```

Comment:
Is this reasonable given that there is an `around` block wrapping the whole example that does have a safe mode-compliant block call?

Reply:
👍